### DISCUSSION TITLE
请教一下track-list.lua如何实现单键显示/关闭轨道菜单

### DISCUSSION THREAD

**Discussion by redomCL** [dyphire/mpv-config#50] [Q&A] [answered]
不好意思，我不是很懂lua脚本，目前你的track-list是按一个对应按键显示一种轨道，然后按esc才能关闭，我想实现一个按键按一下显示，再按一下关闭菜单，就像你的chapter-list.lua一样，想请教一下如何实现，是需要修改局部代码，还是需要大幅度重写。非常感谢

我好像误点了问题解决，其实还没解决。。。。

**Comment by dyphire** [comment, marked as answer]
在最新的提交 https://github.com/dyphire/mpv-config/commit/1822e120ec9d103a18c1c292fd74399051246876 中实现

**Reply by redomCL** [reply]
非常感谢，问题解决！

**Reply by redomCL** [reply]
你好大佬，另外我发现这个轨道菜单不能像lav滤镜一样显示音频位深，请问能否支持显示
![0](https://github.com/dyphire/mpv-config/assets/22269254/8f2c52c2-0f4f-435a-96f8-f50e8f955da4)

**Reply by dyphire** [reply]
mpv 本身不提供位深属性，无法实现

**Reply by redomCL** [reply]
好的知道了，感谢答复

**Reply by redomCL** [reply]
![QQ图片20240229174116](https://github.com/dyphire/mpv-config/assets/22269254/94e042ea-2b65-4a2f-90b0-de39c3372371)
大佬码率长度超过3位数会变成这种形式的显示，154行左右的问题，但我没会改

**Reply by redomCL** [reply]
![2](https://github.com/dyphire/mpv-config/assets/22269254/551578d2-c4b5-4030-a7ec-62ac05d0d63e)
看懂了，请大佬改下码率显示长度
if track['demux-bitrate'] then h(string.format('%.5g kbps', track['demux-bitrate'] / 1000)) end

**Reply by dyphire** [reply]
没什么用，这是 mpv 某次 pr 调整了内部浮点精度导致会出现这种科学计数，即使是内部的 osd 统计信息也会出现这种情况。你只是增加一位显示精度，并不能完全避免它

**Reply by redomCL** [reply]
主要是有些蓝光原盘的音频码率上千了，加到5位可以正常显示千位数的码率和23.976fps这种帧数规格，不从技术上讲，只从普通用户角度讲用来查看视频信息应该已经足够了。

**Reply by dyphire** [reply]
试试最新提交有没有解决这个问题

**Reply by redomCL** [reply]
抱歉有点强迫症了，按照以往常见的习惯，我觉得视频帧数显示到小数点后三位比较好（23.976），%.3f，然后码率显示到整数就好，总之谢谢大佬，我知道怎么设置精度了。问题解决！
